File: day02/py/part1.py
#Read data 
text_file = open("../input.txt", "r")
lines = text_file.readlines()

#lines=["2x3x4","1x1x10"]

mytotal=0
for line in lines:
	thisdim=list(map(int,line.strip("\n").split("x")))
	firstdim=thisdim[0]*thisdim[1]
	seconddim=thisdim[1]*thisdim[2]
	thirddim=thisdim[0]*thisdim[2]
	minall=min(firstdim,seconddim,thirddim)
	thisone=(2*firstdim+2*seconddim+2*thirddim+minall)
	#print(firstdim,seconddim,thirddim,minall,thisone)
	mytotal+=thisone
	

print("Final:",mytotal)

File: day02/js/part1.js
//Read Data
fs  = require("fs");
array = fs.readFileSync("../input.txt").toString().split('\n');
//console.log(array);

mytotal=0
for (i = 0; i < array.length-1; i++) {//for some reason an extra blank element is being picked up
	thisdim=array[i].split("x");
	//console.log(thisdim)
	
	firstdim=parseInt(thisdim[0])*parseInt(thisdim[1])
	seconddim=parseInt(thisdim[1])*parseInt(thisdim[2])
	thirddim=parseInt(thisdim[0])*parseInt(thisdim[2])
	minall=Math.min(firstdim,seconddim,thirddim)
	thisone=(2*firstdim+2*seconddim+2*thirddim+minall)
	
	mytotal+=thisone
}	

console.log("Final:",mytotal)